### ISSUE TITLE
Issue with FileUploader and debug setting

### ISSUE THREAD

**Issue by KamilGrudz** [dddomodossola/remi#291]
When creating the application for data conversion purposes, I noticed that uploading files in a special case does not behave correctly in FileUploader. I am not sure if I did not do something incorrectly, so I wrote a small application only with a upload button with similar functionality as in my application, and the problem remained.

The problem appears while simultaneously:
- using the function set_on_success_listener with FileUploader,
- using the function set_on_data_listener with FileUploader,
- setting "debug = True" on REMI application startup.

I am using the set_on_success_listener function and the set_on_data_listener function only to count the number of uploaded files and locking the other buttons when uploading files. The upload concerns the ESRI Shapefiles, but I also tested the download on ordinary text files and the problem remained.

The problem is:
- the contents of the set_on_success_listener and set_on_data_listener functions are not executed in random cases.
- some files are not uploaded to the indicated folder.

If it is useful, I can provide the source code that does not work properly. I can also describe the problem in greater detail, if it is needed.

**Comment by dddomodossola** [comment]
@KamilGrudz yes please show me your code, I will test it soon. Thank you for reporting. Furthermore, please tell me the python version and browser type/version.

**Comment by KamilGrudz** [comment]
@dddomodossola Thank you for the quick reply.

Python version: 2.7.14 (v2.7.14: 84471935ed, Sep 16 2017, 20:25:58) [MSC v.1500 64 bit (AMD64)]
Browser type: Google Chrome
Browser version: 73.0.3683.103 (64-bit)

I am sending you two python files. Both are used for the same counting of upload files, but they approach it in different ways. In both of them, I added the print function to better show where the problem lies.
The value of the countAll variable should count the execution of the set_on_data_listener function. The countControl value should count the execution of the set_on_success_listener function.
The script should display the word DONE if it finishes uploading files.

The first TestOne.py script does not upload all files and does not count them correctly.
The second TestTwo.py script upload all files, but does not count them correctly.

[Tests.zip](https://github.com/dddomodossola/remi/files/3073421/Tests.zip)

**Comment by dddomodossola** [comment]
I tested the TestOne.py without problems. Or maybe I'm misunderstanding your problem. After Each file upload completes, variables **countAll** and **countControl** are assigned to 0, so every time an upload happen their value is 1. Uploads are executed correctly. Maybe are you working on files opened by other processes?

**Comment by KamilGrudz** [comment]
The problem occurs when uploading multiple files at once. My test concerned over 100 small files on one upload, a few kilobits each.

Files are not open in other applications.

**Comment by dddomodossola** [comment]
Ok, I haven't tested this way. I will further investigate. However compliments for your code, I liked how it's written

**Comment by KamilGrudz** [comment]
Thank you very much. The main app code is better organized. It consists of several thousand lines of code, so I have to keep it clean :) I hope you will be able to reproduce issue. I will not be able to reply to messages over the weekend, so, if anything, I will write back on Monday. Have a nice weekend.

**Comment by dddomodossola** [comment]
Have a nice weekend too. I'm still unable to reproduce the problem with 200 files..

```
countAll:` 200, countControl: 197

remi.server.ws   DEBUG    send_message: 3... -> ('127.0.0.1', 61296)
remi.server.ws   DEBUG    on_message: callback
countAll: 200, countControl: 198

remi.server.ws   DEBUG    send_message: 3... -> ('127.0.0.1', 61296)
remi.server.ws   DEBUG    on_message: callback
countAll: 200, countControl: 199

remi.server.ws   DEBUG    send_message: 3... -> ('127.0.0.1', 61296)
remi.server.ws   DEBUG    on_message: callback
countAll: 200, countControl: 200

DONE-----------------

countSHP: 0
```

All seems to be correct

**Comment by KamilGrudz** [comment]
Thank you for your time and effort.
I will try to re-install Remi and run the script on different machine.

**Comment by KamilGrudz** [comment]
I also have some questions about the progress bar in Remi, but I do not know whether to write them here or, for the sake of order, start a new post with an adequate title?

**Comment by dddomodossola** [comment]
please create separate threads for different topics ;-)

**Comment by KamilGrudz** [comment]
I tried to reinstall REMI, but unfortunately it did not help. I still must try to test the script on another machine. For now, disabling the debug option solves the problem.
